### ISSUE TITLE
Specs for `TextStyleFactory` and `TextSegment`

### ISSUE THREAD

**Issue by jasonrclark** [shoes/shoes4#1097]
Interactions with styling aren't well spec'd, and in moving things around some coverage was removed in 406e925.

I don't necessarily want precisely those specs back, but would rather we test the classes appropriately for how we think they should be tested today.

Newcomer friendly since those classes are pretty isolated, so just backfilling a spec for it should be relatively accessible.

**Comment by piisalie** [comment]
Is this still wanted? I did a little git digging on the TextSegment and TextSegmentCollection specs which look like they haven't been updated much since this issue. I didn't see any TextStyleFactory specs. (I could be looking in the wrong place / misunderstanding)

I'd be willing to give this a try if it still needs doing.

**Comment by jasonrclark** [comment]
Yeah, this is definitely still desired. That area isn't well tested, so any help there could be appreciated.

`TextStyleFactory` is also a good candidate from what I remember for some more testing. The struggle there I think will be getting specs that aren't just transliterating the production code, but worth see what we can do.

Thanks!
